src/soc-support.md

```diff
@@ -0,0 +1,183 @@
+issues related to register description files or other resources. The use of a
+public issue tracking system (like GitHub Issues) for reporting and tracking
+problems might help. Actively engage with the community through forums,
+discussions, and updates to build trust and collaboration.
+
+## Maintaining PAC and HAL Crates
+
+Peripheral Access Crates (PACs) and Hardware Abstraction Layer (HAL) crates are
+at the core of enabling Rust support.
+
+### Generate and Maintain PACs
+
+Multiple tools such as `svd2rust`, `chiptool`, `raltool`, and `svd2pac`
+automate the generation of PACs from register description files. Each tool has
+its strengths, and selecting the right one depends on the requirements and the
+complexity of the hardware.
+
+### Develop and Maintain HAL Crates
+
+Implement `embedded-hal` and `embedded-hal-async` traits in your HAL crates.
+Adhering to these traits ensures compatibility across the Embedded Rust
+ecosystem, enhancing interoperability. It is an essential goal that HALs use
+Rust code rather than wrapping existing C code. An incremental porting
+strategy, where Rust is used for all core functionality, but C with Rust
+bindings is used for complex drivers, is acceptable, allowing for gradual
+adoption and community contributions.
+
+Start with essential peripherals (clock, timer, GPIO) and expand progressively
+(I2C, SPI, UART, etc.) based on community feedback. Release early and often to
+engage the community and gather valuable insights for further development.
+
+### Common Recommendations
+
+- Ensure that crates are compatible with `no_std` environments, which are
+  common in embedded systems without an operating system. Functionality that
+  needs `alloc` or `std` can be included when gated with Cargo "features."
+- Make your crates available on [crates.io](https://crates.io/) to maximize
+  visibility and ease of use for developers.
+- Use semantic versioning to maintain consistency and predictability in your
+  releases.
+- Prefer licenses like Apache 2.0 and MIT for their permissive nature, which
+  encourages broader adoption and collaboration.
+
+### Issue Tracking
+
+Effective issue tracking is crucial for maintaining a healthy and collaborative
+ecosystem. Discuss triaging, labeling, and community involvement in issue
+resolution. Implement transparent processes for:
+
+- Triage and prioritize issues based on severity and impact.
+- Use labels to categorize issues (e.g., bugs, feature requests).
+- Encourage community members to contribute to resolving issues by providing
+  feedback or submitting pull requests (PRs).
+
+### Facilitate Debugging and Testing
+
+The use of `probe-rs` is prevalent in the Embedded Rust ecosystem for debugging
+and testing. Combined with debug-based facilities like `defmt-rtt`, which
+offers logging capabilities for embedded systems, the Embedded Rust ecosystem
+has developed numerous tools. `probe-rs` supports a wide range of target
```

Comment:
the final sentence here --- the Embedded Rust ecosystem has developed numerous tools --- doesn't seem to fit well in the paragraph. How about?

The Embedded Rust ecosystem offers various tools used for debugging
and testing, with `probe-rs` being one of the most widely
used. Combined with debug-based facilities like `defmt-rtt`, which
provides logging capabilities for embedded systems, these tools form a
powerful foundation for development. `probe-rs` supports a wide range
of target architectures, debug interfaces, and debug probe protocols.

Thorough testing ensures hardware-software reliability, and leveraging these
tools can significantly enhance development workflows.